Load Code

PaShkadun opened the pull request:
Load Code

Commit:
Load Code

PaShkadun commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[TMM-22] Princess Game](https://trello.com/c/Hn3SBy26/54-tmm-22-princess-game)

AnastasiaKorotkevich reviewed (commented):
Не соблюдена конвенция.

Commit:
Relize code. Edit Convention.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+                position[i] = random.Next(1, 99);
+                for (int o = 0; o < i; o++)
+                {
+                    if (position[i] == position[o]) { i--; }
```
Конвенция не соблюдена.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+        {
+            Mine mine = new Mine(0);
+            //Если юзер на клетке 9-9 - победа
+            if (player.PlayerPositionHorizontal == 9 && player.PlayerPositionVertical == 9) { player.SetHpWin(); return 0; }
+            //Если клетка пустая, то ничего не происходит
+            if (cell[player.PlayerPositionHorizontal][player.PlayerPositionVertical] == null) { return 0; }
+            //Если на клетке мина
+            if (cell[player.PlayerPositionHorizontal][player.PlayerPositionVertical].GetType() == mine.GetType())
+            {
+                mine = (Mine)cell[player.PlayerPositionHorizontal][player.PlayerPositionVertical];
+                if (mine.Status == "Active") { mine.InactiveMine(); return mine.Damage; } //Если активная, снимает HP
+                else { return 0; }
+            }
+            else { return 0; }
```
Конвенция не соблюдена.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+
+namespace Shkadun_Princess
+{
+    class Map
```
Модификатор.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+{
+    class Map
+    {
+        object[][] cell;            //карта
```
Тип выбран неправильно.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+    class Map
+    {
+        object[][] cell;            //карта
+        WorkWithConsole workWithConsole;   //Объект класса для консольного ввода-вывода сообщений
```
Название не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+        //Создание игры
+        public void StartNewGame(Player player)
+        {
+            cell = null;
```
Здесь это не нужно.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+        {
+            cell = null;
+            cell = new object[10][];
+            for (int i = 0; i < 10; i++)
```
Что такое 10?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+        }
+
+        //Генерация мин
+        public void GenerationMines()
```
Название не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+            Random random = new Random();
+            int[] position = new int[10];
+            int damage;
+            for (int i = 0; i < 10; i++)
```
Что такое 10?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+            for (int i = 0; i < 10; i++)
+            {
+                position[i] = random.Next(1, 99);
+                for (int o = 0; o < i; o++)
```
Что за о?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+            int damage;
+            for (int i = 0; i < 10; i++)
+            {
+                position[i] = random.Next(1, 99);
```
Что такое 1, 99?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+                    if (position[i] == position[o]) { i--; }
+                }
+                damage = random.Next(1, 10);
+                cell[position[i] / 10][position[i] % 10] = new Mine(damage);    //Положение мины
```
Что такое 10?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+        }
+
+        //Проверка мины на клетке, на которую стал юзер
+        public int CheckMine(Player player)
```
Название не совсем походит.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+        //Проверка мины на клетке, на которую стал юзер
+        public int CheckMine(Player player)
+        {
+            Mine mine = new Mine(0);
```
Что такое 0?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+            if (cell[player.PlayerPositionHorizontal][player.PlayerPositionVertical].GetType() == mine.GetType())
+            {
+                mine = (Mine)cell[player.PlayerPositionHorizontal][player.PlayerPositionVertical];
+                if (mine.Status == "Active") { mine.InactiveMine(); return mine.Damage; } //Если активная, снимает HP
```
Захардкоженая строка.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+                for (int o = 0; o < cell[i].Length; o++)
+                {
+                    //Если положение юзера, выводим Y
+                    if (i == player.PlayerPositionHorizontal && o == player.PlayerPositionVertical) { workWithConsole.DrowPlayer(); }
+                    //Если пустая, то пустую ячейку
+                    else if (cell[i][o] == null) { workWithConsole.DrowEmptyCell(); }
+                    //Если мина, то проверяем её статус
+                    else if (cell[i][o].GetType() == mine.GetType()) { workWithConsole.DrowMineCell((Mine)cell[i][o]); }
```
Конвенция не соблюдена.

AnastasiaKorotkevich replied:
Обязательно ли использовать GetType?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+            workWithConsole.DrowLine(10);
+        }
+
+        public Map()
```
Название не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+                }
+                Console.WriteLine();
+            }
+            workWithConsole.DrowLine(10);
```
Что такое 10?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Map.cs:
```diff
@@ -0,0 +1,94 @@
+        {
+            workWithConsole = new WorkWithConsole();
+            cell = new object[10][];
+            for (int i = 0; i < 10; i++)
```
Что такое 10?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Mine.cs:
```diff
@@ -0,0 +1,20 @@
+﻿
+namespace Shkadun_Princess
+{
+    class Mine
```
Модификатор.

AnastasiaKorotkevich replied:
Если это класс Mine, то почему генерация мин происходит в другом классе? Неправильная абстракция.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Mine.cs:
```diff
@@ -0,0 +1,20 @@
+
+        public void InactiveMine()
+        {
+            Status = "Inactive";
+        }
+
+        public Mine(int damage)
+        {
+            Status = "Active";
+            Damage = damage;
```
Хардкод. Захардкоженые строки выносят в константы.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+﻿
+namespace Shkadun_Princess
+{
+    class Player
```
Модификатор.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+    class Player
+    {
+        public int HP { get; private set; }
+        public int PlayerPositionHorizontal { get; private set; }
+        public int PlayerPositionVertical { get; private set; }
```
Название можно упростить.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+        //Для проверки победы. Если hp > 10, то в program выводит победу.
+        public void SetHpWin()
+        {
+            HP = 11;
```
Что такое 11?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+            HP = 11;
+        }
+
+        public void StartPlayerPosition()
+        {
+            PlayerPositionHorizontal = 0;
+            PlayerPositionVertical = 0;
+            HP = 10;
+        }
```
Есть ли смысл в этом методе?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+            HP = 10;
+        }
+
+        public void PlayerRunUp(Map map)
```
Название не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+        public void PlayerRunUp(Map map)
+        {
+            //Если юзер находится на верхней ячейке
+            if (PlayerPositionHorizontal == 0) { return; }
```
Не по конвенции.

AnastasiaKorotkevich replied:
Что такое 0?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+            //Если юзер находится на верхней ячейке
+            if (PlayerPositionHorizontal == 0) { return; }
+
+            PlayerPositionHorizontal -= 1;
```
Что такое 1?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+            HP -= damage;                     //Снимает 0 HP
+            map.DrowMap(this);
+        }
+        public void PlayerRunDown(Map map)
```
Отступ. Название не по конвенции. Дублирование кода.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+        public void PlayerRunDown(Map map)
+        {
+            //Если юзер находится на нижней клетке
+            if (PlayerPositionHorizontal == 9) { return; }
```
Не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+            //Если юзер находится на нижней клетке
+            if (PlayerPositionHorizontal == 9) { return; }
+
+            PlayerPositionHorizontal += 1;
```
То же самое.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+            HP -= damage;                     //Снимает 0 HP
+            map.DrowMap(this);
+        }
+        public void PlayerRunRight(Map map)
+        {
+            //Если юзер на правой ячейке
+            if (PlayerPositionVertical == 9) { return; }
+
+            PlayerPositionVertical += 1;
+            int damage = map.CheckMine(this); //Проверяет, стал ли юзер на мину. Если нет
+            HP -= damage;                     //Снимает 0 HP
+            map.DrowMap(this);
+        }
+        public void PlayerRunLeft(Map map)
+        {
+            //Если юзер на левой ячейке
+            if (PlayerPositionVertical == 0) { return; }
+
+            PlayerPositionVertical -= 1;
+            int damage = map.CheckMine(this); //Проверяет, стал ли юзер на мину. Если нет
+            HP -= damage;                     //Снимает 0 HP
+            map.DrowMap(this);
+        }
```
Отступ. Название не по конвенции. Дублирование кода. Что такое 1 и 0? 
Конвенция не соблюдена.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Player.cs:
```diff
@@ -0,0 +1,71 @@
+            map.DrowMap(this);
+        }
+
+        public Player()
+        {
+            HP = 10;
+            PlayerPositionHorizontal = 0;
+            PlayerPositionVertical = 0;
+        }
```
Дублирование кода.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Program.cs:
```diff
@@ -1,12 +1,45 @@
-namespace Shkadun_Homeworks
+namespace Shkadun_Princess
 {
     class Program
```
Модификатор.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Program.cs:
```diff
@@ -1,12 +1,45 @@
-        { 
-            Console.WriteLine("Hello World!");
+        {
+            WorkWithConsole workWithConsole = new WorkWithConsole();
```
Название класса не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Program.cs:
```diff
@@ -1,12 +1,45 @@
+            map.GenerationMines();
+            map.DrowMap(player);
+
+            while (true)
```
Точно нужно юзать while(true)?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Program.cs:
```diff
@@ -1,12 +1,45 @@
+            {
+                switch (Console.ReadKey(true).Key)
+                {
+                    case ConsoleKey.W:
+                    case ConsoleKey.UpArrow: player.PlayerRunUp(map); break;
+                    case ConsoleKey.S:
+                    case ConsoleKey.DownArrow: player.PlayerRunDown(map); break;
+                    case ConsoleKey.D:
+                    case ConsoleKey.RightArrow: player.PlayerRunRight(map); break;
+                    case ConsoleKey.A:
+                    case ConsoleKey.LeftArrow: player.PlayerRunLeft(map); break;
+                    default: break;
```
Не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Program.cs:
```diff
@@ -1,12 +1,45 @@
+                }
+                if (player.HP <= 0)
+                {
+                    if (workWithConsole.GameOver(1) == 0) { break; }
```
Не по конвенции.

AnastasiaKorotkevich replied:
Что такое 1 и 0?

AnastasiaKorotkevich commented on Shkadun_Homeworks/Program.cs:
```diff
@@ -1,12 +1,45 @@
+                }
+                else if (player.HP > 10)
+                {
+                    if (workWithConsole.GameOver(0) == 0) { break; }
```
Не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/Program.cs:
```diff
@@ -1,12 +1,45 @@
+                    if (workWithConsole.GameOver(0) == 0) { break; }
+                    map.StartNewGame(player);
+                }
+                Thread.Sleep(200);
```
Для чего это?

AnastasiaKorotkevich commented on Shkadun_Homeworks/WorkWithConsole.cs:
```diff
@@ -0,0 +1,51 @@
+
+namespace Shkadun_Princess
+{
+    class WorkWithConsole
```
Модификатор.

AnastasiaKorotkevich commented on Shkadun_Homeworks/WorkWithConsole.cs:
```diff
@@ -0,0 +1,51 @@
+    {
+        public int GameOver(int result)
+        {
+            if (result == 0) { Console.WriteLine("Вы выиграли."); }
+            else { Console.WriteLine("Вы проиграли. Ещё раз?"); }
+
```
Не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/WorkWithConsole.cs:
```diff
@@ -0,0 +1,51 @@
+    {
+        public int GameOver(int result)
+        {
+            if (result == 0) { Console.WriteLine("Вы выиграли."); }
+            else { Console.WriteLine("Вы проиграли. Ещё раз?"); }
```
Захардкоженые строки.

AnastasiaKorotkevich commented on Shkadun_Homeworks/WorkWithConsole.cs:
```diff
@@ -0,0 +1,51 @@
+            if (result == 0) { Console.WriteLine("Вы выиграли."); }
+            else { Console.WriteLine("Вы проиграли. Ещё раз?"); }
+
+            Console.WriteLine("Ещё раз? y/n");
```
Захардкоженая строка.

AnastasiaKorotkevich commented on Shkadun_Homeworks/WorkWithConsole.cs:
```diff
@@ -0,0 +1,51 @@
+
+            Console.WriteLine("Ещё раз? y/n");
+
+            if (Console.ReadLine() == "y") { return 1; }
+            else { return 0; }
```
Что такое y? 1? 0?
Не по конвенции.

AnastasiaKorotkevich commented on Shkadun_Homeworks/WorkWithConsole.cs:
```diff
@@ -0,0 +1,51 @@
+        {
+            Console.WriteLine("----------------------");
+
+            if (line < 10) { Console.Write("|"); }
```
Не по конвенции. 
Здесь метод отрисовки поля? Неправильная абстракция.

AnastasiaKorotkevich commented on Shkadun_Homeworks/WorkWithConsole.cs:
```diff
@@ -0,0 +1,51 @@
+            if (line < 10) { Console.Write("|"); }
+        }
+
+        public void DrowPlayer()
+        {
+            Console.Write("Y|");
+        }
+
+        public void DrowEmptyCell()
+        {
+            Console.Write(" |");
+        }
+
+        public void DrowMineCell(Mine mine)
+        {
+            if (mine.Status == "Active") { Console.Write(" |"); }
+            else { Console.Write("X|"); }
+        }
```
Не по конвенции. 
Здесь метод отрисовки поля? Неправильная абстракция.

AnastasiaKorotkevich commented on Shkadun_Homeworks/WorkWithConsole.cs:
```diff
@@ -0,0 +1,51 @@
+            else { Console.Write("X|"); }
+        }
+
+        public void WriteHP(Player player)
+        {
+            Console.WriteLine($"HP {player.HP}");
+        }
+
+        public void WriteGameName()
+        {
+            Console.WriteLine("The Princess game");
+        }
```
Есть ли смысл в этих методах? Захардкоженые строки.

Commit:
Edit convention misstakes and edit files "map.cs" and "WorkWithConsole.cs"(renamed in "ConsoleWork.cs").

BitW13 reviewed (commented):
Говно. Ошибки не исправлены

Commit:
Edit mistakes previous commits. I don't saw they :(
1. Edit Map.cs(renamed in "Game.cs"
2. Edit Player.cs(deleted methods Run, add method Move)
3. Edit Program.cs(edit switch construction)
4. Edit ConsoleWork.cs

BitW13 reviewed (commented):
Говно. Пересмотреть видео и переделать

Commit:
Edit convention mistakes in Game.cs, Mine.cs, Player.cs, Program.cs. Again.

Commit:
Edit error in program.

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,137 @@
+
+                return 0; 
+            }
+
```
Лишний отступ

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,137 @@
+            { 
+                return 0;
+            }
+
```
Лишний отступ

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,137 @@
+{
+    public class Game
+    {
+        private const string WIN = "Win";
```
Название констант не по конвенции

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,137 @@
+            {
+                player.GameOver = WIN;
+
+                return 0; 
```
Что такое 0?

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,137 @@
+
+            else if (gameField[player.PositionVertical][player.PositionHorizontal] == null) 
+            { 
+                return 0;
```
Что такое 0?

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,137 @@
+                        }
+                        else
+                        {
+                            return 0;
```
Что такое 0?

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,137 @@
+                    }
+                }
+
+                return 0; 
```
Что такое 0?

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,137 @@
+            }
+            else 
+            { 
+                return 0; 
```
Что такое 0?

BitW13 commented on Shkadun_Homeworks/Program.cs:
```diff
@@ -1,12 +1,67 @@
+                        player.Move(game, horizontal: -1); 
+                        break;
+
+                    default: break;
```
На новую строчку

Commit:
Edit const names, switch construction in Program.cs and replace 0 in Game.cs.

AnastasiaKorotkevich reviewed (commented):
Не все ошибки исправлены.

Commit:
Delete excess spaces. Edit 0 in Game.cs.

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,133 @@
+        private const int NullDamage = 0;
+
+        string[][] gameField;
+        private int countMines = 10;
```
А зачем тебе эта переменная, если у тебя есть List<Mine>, из которого можно получить count?
И зачем тебе вообще список, если у тебя известное количество мин (10)? тогда массив

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,133 @@
+        private int countMines = 10;
+        private List<Mine> ListMines;
+
+        public int CheckCell(Player player)
```
Зачем каждый раз пробрасывать игрока, если в игре может быть только один игрок?

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,133 @@
+
+        string[][] gameField;
+        private int countMines = 10;
+        private List<Mine> ListMines;
```
camelCase

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,133 @@
+    {
+        private const string Win = "Win";
+        private const string Bomb = "Bomb";
+        private const string Active = "Active";
```
Active и Inactive в enum

BitW13 commented on Shkadun_Homeworks/Game.cs:
```diff
@@ -0,0 +1,133 @@
+            return damage;
+        }
+
+        public void DrowMap(Player player)
```
DrawMap

Commit:
Edit ListMines, names of vars, DrowMap(renamed in DrawMap), delete Player in Program.cs and create in Game.cs, create StatusBomb enum.

Commit:
Delete const countMines and add in-param in Game construct.

BitW13 reviewed (approved):
Мержи
